src/main/java/org/mozilla/jss/crypto/MessageDigestProvider.java

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.jss.crypto;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Chris S. Kelley
+ *
+ * Class to provide MessageDigests and utility methods to work with them
+ * Primarily created to track use of deprecated algorithms
+ */
+public class MessageDigestProvider {
```

Comment:
The issue is that we will only see the warnings if we modify the code to use this class. Other code that still use the standard Java security classes will not generate warnings.

I was wondering if we should add a `JSSProvider.getService()` and generate the warnings there instead:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/Provider.html#getService(java.lang.String,java.lang.String)

See also:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/security/MessageDigest.java#L179-L201
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/sun/security/jca/GetInstance.java#L152-L182
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/security/Provider.java#L1228-L1250

Reply:
Good point, I have given it a go.

Sorry for the diff pollution, undoing the `MessageDigestProvider` stuff left lots of whitespace/import tidying.